src/main.cpp

```diff
@@ -438,15 +438,33 @@ int main(int argc, char* argv[])
     const bool firstInstanceOfMudlet = instanceCoordinator->tryToStart();
 
     const QStringList positionalArguments = parser.positionalArguments();
+    QString telnetUri;
     if (!positionalArguments.isEmpty()) {
-        const QString absPath = QDir(positionalArguments.first()).absolutePath();
-        instanceCoordinator->queuePackage(absPath);
-        if (!firstInstanceOfMudlet) {
-            const bool successful = instanceCoordinator->installPackagesRemotely();
-            if (successful) {
-                return 0;
-            } else {
-                return 1;
+        const QString firstArg = positionalArguments.first();
+        // Check if it's a telnet:// URI
+        if (firstArg.startsWith(qsl("telnet://"))) {
+            telnetUri = firstArg;
+            // If another instance is running, send the URI to it
+            if (!firstInstanceOfMudlet) {
+                instanceCoordinator->queuePackage(firstArg);
+                const bool successful = instanceCoordinator->installPackagesRemotely();
+                if (successful) {
+                    return 0;
+                } else {
```

Comment:
Minor detail - but this is an "else-after-return" - so the else isn't needed - and in fact the whole `if`can be replaced by: `return successful ? 1 : 0` !